src/anemoi/datasets/create/input.py

```diff
@@ -108,14 +108,15 @@ def _data_request(data):
     for field in data:
         try:
             if date is None:
-                date = field.datetime()["valid_time"]
+                date = field.metadata("valid_datetime")
 
-            if field.datetime()["valid_time"] != date:
+            if field.metadata("valid_datetime") != date:
```

Comment:
What is this line doing?

Reply:
i.e. why continue?

Reply:
We only care about one date. So we consider only the fields with the same valid date as the first.